### ISSUE TITLE
CMake build system

### ISSUE THREAD

**Issue by nemo1369** [boostorg/multiprecision#139]
Well, first of all, I'd like to ask, are there some chance for the CMake-based build system to be merged?

I mean, I'm using it for the development, so I think that would be useful for others to have an optional support for it too.

**Comment by jzmaddock** [comment]
I'm not a CMake user, and don't speak the language as it were, so while I'm not totally against it, such a submission would have to come with at least some minimal tests (hooked up to the CI system) to verify that it does actually work and pull in all the dependencies it needs.

**Comment by nemo1369** [comment]
That is great. I have just the stuff. Pull request is coming.

**Comment by nemothenoone** [comment]
I believe it would be great to link this issue to PR #143, so it will close just as that PR will be merged.

**Comment by NAThompson** [comment]
@nemothenoone : Is there a chance that the `CMake` system can produce a C++-20 module for Boost.Multiprecision, without jeopardizing backwards compatibility for users who consume the header-only version?

**Comment by nemothenoone** [comment]
@NAThompson Definitely. Kitware guys already has modules support implementation going, there is an example available: https://github.com/mathstuf/cxx-modules-sandbox. So, I'ill simply introduce same mechanism usage in here.

**Comment by NAThompson** [comment]
> Or are you talking about generating the module definition itself?

Sadly I haven't yet educated myself enough on how to do this to even understand this comment! What I had envisioned was that CMake would somehow make putting a module up on conan.io easier . . .
